src/devices/Card/CreditCard/TagDetails.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Iot.Device.Card.CreditCardProcessing
+{
+    /// <summary>
+    /// Class containing the Tag and the details including description
+    /// </summary>
+    public class TagDetails : Tag
+    {
+        public TagDetails()
+        { }
+
+        /// <summary>
+        /// Constructor using an existing Tag
+        /// </summary>
+        /// <param name="tag"></param>
+        public TagDetails(Tag tag)
```

Comment:
would it make sense to replace this with Clone() and implement IClonable? Is this actually used anywhere?

Reply:
I'm using the tag details mainly with the static list containing most of the tag I've been able to find in the specs. I don't necessary need to clone it.